### PR TITLE
Add basic docs

### PR DESCRIPTION
**Pull request by ZedThree**
And add the generated docs

### CODE REVIEW

File: fortitude_dev/src/main.rs [hunk 1]
```diff
@@ -0,0 +1,51 @@
+//! This crate implements an internal CLI for developers of Fortitude.
+//!
+//! Within the fortitude repository you can run it with `cargo dev`.
```
Comment (LiamPattinson) [inline fortitude_dev/src/main.rs:3]: I couldn't get this to work, but going into `fortitude_dev` and using `cargo run` was fine. Please could we update this here?

**Review by LiamPattinson — CHANGES_REQUESTED** [review]
This all looks great! Just two things to address besides the review comment:
- Do we need both `docs/all_rules.md` and `docs/rules/rules.md`? As far as I can tell they're duplicates. Also, would it be possible to have these auto-generated with the `generate-docs` command?
- Yeah, I think it would be worth removing the commented-out code for now.

**Comment by LiamPattinson** [comment]
Another thing, any chance of a CI workflow that checks that the docs are up-to-date (or possibly auto-updates them)? I could try to add that in a future PR if you'd prefer.

**Comment by ZedThree** [comment]
Ok, I pulled my finger out and made some proper(ish) docs.

**Comment by ZedThree** [comment]
Built docs here: https://fortitude.readthedocs.io/en/dev-command/

File: fortitude_dev/src/main.rs (hunk 1 above)
Comment (LiamPattinson) [reply fortitude_dev/src/main.rs:3]: `cargo dev` is working now, thanks!

File: README.md [hunk 2]
```diff
@@ -180,6 +180,10 @@ rules on top of those in the configuration file.
 fortitude check --extend-select=M
 ```
 
+## Documentation
+
+See [table of rules](docs/all_rules.md) for a list of all rules.
```
Comment (LiamPattinson) [inline README.md:185]:
```suggestion
See [table of rules](docs/rules.md) for a list of all rules.
```

**Review by LiamPattinson — COMMENTED** [review]
Wow, the docs look fantastic! Is there a way to build the html locally?

**Comment by ZedThree** [comment]
Yup:

```console
$ uv pip install mkdocs-material
$ mkdocs serve
```

This watches for file changes and reloads the site as needed.

File: README.md (hunk 2 above)
Comment (ZedThree) [reply README.md]: Pointed this to readthedocs instead (although link won't work until this is merged)

**Comment by LiamPattinson** [comment]
> Yup:
> 
> ```
> $ uv pip install mkdocs-material
> $ mkdocs serve
> ```
> 
> This watches for file changes and reloads the site as needed.

Amazing, works perfectly (and unlike Sphinx, I don't have to type out `file:///the/entire/path/to/my/index.html` in my browser to see the changes).

Please could you add this to `README.dev.md`?

````markdown
## Building Docs

The documentation can be built locally using:

```bash
pip install mkdocs-material
mkdocs serve
```
````

**Review by LiamPattinson — APPROVED** [review]

**Pull request merged**